src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitor.scala

```diff
@@ -49,63 +51,64 @@ object SubmissionMonitor {
  */
 class SubmissionMonitor(workspaceName: WorkspaceName,
                         submissionId: String,
-                        containerDAO: GraphContainerDAO,
-                        datasource: DataSource,
+                        datasource: SlickDataSource,
                         workflowPollInterval: Duration,
                         submissionPollInterval: Duration,
                         workflowMonitorProps: (ActorRef, WorkspaceName, String, Workflow) => Props) extends Actor {
   import context._
+  import datasource.dataAccess.driver.api._
 
   setReceiveTimeout(submissionPollInterval)
-  startWorkflowMonitorActors()
+  startWorkflowMonitorActors() pipeTo self
 
   override def receive = {
     case WorkflowStatusChange(workflow, workflowOutputs) =>
       system.log.debug("workflow state change, submission {}, workflow {} {}", submissionId, workflow.workflowId, workflow.status)
-      handleStatusChange(workflow, workflowOutputs)
+      handleStatusChange(workflow, workflowOutputs) pipeTo self
```

Comment:
all these functions now return futures so pipe them to self so errors are not lost